File: web-sample/src/main/java/pl/psi/aaas/RestEndpoint.java
package pl.psi.aaas;

import java.time.ZonedDateTime;
import java.util.HashMap;
import java.util.Map;

import javax.inject.Inject;
import javax.ws.rs.GET;
import javax.ws.rs.Path;

import pl.psi.aaas.usecase.CalculationDefinition;
import pl.psi.aaas.usecase.CalculationDefinitionIf;

@Path("/call")
public class RestEndpoint {

    private final Facade< CalculationDefinitionIf, Void> facade;

    @Inject
    public RestEndpoint(Facade< CalculationDefinitionIf, Void> facade) {
        this.facade = facade;
    }

    @GET
    public void callDef() {
        facade.callScript(getDefinition());
    }

    public CalculationDefinitionIf getDefinition() {
        String[] strArr = {"a", "b", "c"};
        ZonedDateTime[] dtArr = {ZonedDateTime.now().minusHours(2), ZonedDateTime.now().minusHours(1), ZonedDateTime.now()};
        Vector<String> strVec = Parameter.ofArrayNotNull(strArr, String.class);
        Vector<ZonedDateTime> dtVec = Parameter.ofArrayNotNull(dtArr, ZonedDateTime.class);
        Vector<Long> longVec = Parameter.ofArrayNotNull(new Long[]{1L, 2L, 3L}, Long.class);
        Vector<Double> doubleVec = Parameter.ofArrayNotNull(new Double[]{0.1, 0.2, 1.0}, Double.class);
        Vector<Double> doubleNullVec = Parameter.ofArray(new Double[]{0.1, null, 1.0}, Double.class);
        Vector<Boolean> boolVec = Parameter.ofArrayNotNull(new Boolean[]{true, false, true}, Boolean.class);
        Vector<Boolean> boolNullVec = Parameter.ofArray(new Boolean[]{true, false, null}, Boolean.class);

        Column[] dfColumns = {
                new Column("dt", (Vector<Object>) (Vector<?>) dtVec),
                new Column("longs", (Vector<Object>) (Vector<?>) longVec),
                new Column("doubles", (Vector<Object>) (Vector<?>) doubleVec)
        };

        Map<String, Parameter<?>> parameters = new HashMap<>();
        parameters.put("str", Parameter.ofPrimitive("str_value"));
        parameters.put("dt", Parameter.ofPrimitive(ZonedDateTime.now()));
        parameters.put("d", Parameter.ofPrimitive(0.75));
        parameters.put("l", Parameter.ofPrimitive(10L));
        parameters.put("b", Parameter.ofPrimitive(false));
        parameters.put("strV", strVec);
        parameters.put("dtV", dtVec);
        parameters.put("longV", longVec);
        parameters.put("doubleV", doubleVec);
        parameters.put("doubleNullV", doubleNullVec);
        parameters.put("boolV", boolVec);
        parameters.put("boolNullV", boolNullVec);
        parameters.put("df", Parameter.ofDataFrame(dfColumns));

        return new CalculationDefinition( "add" );
    }
}

File: web-sample/src/main/kotlin/pl/psi/aaas/RestEndpoint.kt
package pl.psi.aaas

import pl.psi.aaas.usecase.CalculationDefinition
import pl.psi.aaas.usecase.CalculationDefinitionDTO
import java.time.ZonedDateTime
import javax.inject.Inject
import javax.ws.rs.GET
import javax.ws.rs.Path

@Path("/call")
class RestEndpoint @Inject constructor(private val facade: Facade<CalculationDefinition, Unit>) {

    @GET
    fun callDef() {
        facade.callScript(getDefinition())
    }

    fun getDefinition(): CalculationDefinition {
        val strArr = arrayOf("a", "b", "c")
        val dtArr = arrayOf(ZonedDateTime.now().minusHours(2), ZonedDateTime.now().minusHours(1), ZonedDateTime.now())
        val strVec = Parameter.ofArrayNotNull(strArr, String::class.java)
        val dtVec = Parameter.ofArrayNotNull(dtArr, ZonedDateTime::class.java)
        val longVec = Parameter.ofArrayNotNull(arrayOf(1, 2, 3L), Long::class.java)
        val doubleVec = Parameter.ofArrayNotNull(arrayOf(0.1, 0.2, 1.0), Double::class.java)
        val doubleNullVec = Parameter.ofArray(arrayOf(0.1, null, 1.0), Double::class.java)
        val boolVec = Parameter.ofArrayNotNull(arrayOf(true, false, true), Boolean::class.java)
        val boolNullVec = Parameter.ofArray(arrayOf(true, false, null), Boolean::class.java)
        val dfColumns = arrayOf(Column("dt", dtVec as Vector<Any>), Column("longs", longVec as Vector<Any>), Column("doubles", doubleVec as Vector<Any>))

        val parameters = mutableMapOf(
                "str" to Parameter.ofPrimitive("str_value")
                , "dt" to Parameter.ofPrimitive(ZonedDateTime.now())
                , "d" to Parameter.ofPrimitive(0.75)
                , "l" to Parameter.ofPrimitive(10L)
                , "b" to Parameter.ofPrimitive(false)
                , "strV" to strVec
                , "dtV" to dtVec
                , "longV" to longVec
                , "doubleV" to doubleVec
                , "doubleNullV" to doubleNullVec
                , "boolV" to boolVec
                , "boolNullV" to boolNullVec
                , "df" to Parameter.ofDataFrame(dfColumns)
        )

        return CalculationDefinitionDTO("add",
                parameters,
                parameters)
    }
}